OCPBUGS#6800: change the MTU in Local Zones deployments (#55327)
openshift/openshift-docs: modules/installation-localzone-generate-k8s-manifest.adoc

<1> For `<installation_directory>`, specify the installation directory that
contains the `install-config.yaml` file you created.

> . Set the MTU size to 1200

Author: ```suggestion
. Set the default Maximum Transmission Unit (MTU) according to the network plugin:
[IMPORTANT]
====
Generally, the Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300, see [AWS Documentation](https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html). 
The Cluster network MTU must be always less than the EC2 MTU to account for the cluster network overlay overhead. The specific overhead is determined by your network plugin:
- OVN-Kubernetes: 100 bytes
- OpenShift SDN: 50 bytes

Check the ['MTU value selection' for more details](https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu)
====
```

> WANR: Steps described on the official doc https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu

> TODO: add a pop-up explaining that it's a requirement of AWS to communicate withing the nodes in Local Zones and regular zones

> > TODO: check the impact on cluster. Make sure we are validating this on the tests.

Author: Need follo-up into those lines

> > TODO: check the impact on cluster. Make sure we are validating this on the tests.

Author: Those items should be resolved with the teams.

> 1300 bytes between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region.

[AWS recommends](https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html) 1300 to communicate with regular zones, we are decreasing the overhead for OVN (100) as per the recommendation on our docs.

> +
> [source,terminal]
> ----
> $ cat <<EOF > manifests/cluster-network-03-config.yml
> apiVersion: operator.openshift.io/v1
> kind: Network
> metadata:
>   name: cluster
> spec:
>   defaultNetwork:
>     ovnKubernetesConfig:
>       mtu: 1200
> EOF
> ----

Author: ```suggestion
** If you are using the `OVN-Kubernetes` network plugin, enter the following command:
+
[source,terminal]
----
$ cat <<EOF > <installation_directory>/manifests/cluster-network-03-config.yml
apiVersion: operator.openshift.io/v1
kind: Network
metadata:
  name: cluster
spec:
  defaultNetwork:
    ovnKubernetesConfig:
      mtu: 1200
EOF
----

** If you are using the `OpenShift SDN` network plugin, enter the following command:
+
[source,terminal]
----
$ cat <<EOF > <installation_directory>/manifests/cluster-network-03-config.yml
apiVersion: operator.openshift.io/v1
kind: Network
metadata:
  name: cluster
spec:
  defaultNetwork:
    openshiftSDNConfig:
      mtu: 1250
EOF
----
```

<1> For `<installation_directory>`, specify the installation directory that
contains the `install-config.yaml` file you created.

. Set the default Maximum Transmission Unit (MTU) according to the network plugin:
+
[IMPORTANT]
====
Generally, the Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300, see [AWS Documentation](https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html). 
The Cluster network MTU must be always less than the EC2 MTU to account for the cluster network overlay overhead. The specific overhead is determined by your network plugin:
- OVN-Kubernetes: 100 bytes
- OpenShift SDN: 50 bytes

> Check the ['MTU value selection' for more details](https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu)

Reviewer: Remove this line, and put these lines in the assembly after the include statement for this module ( https://github.com/openshift/openshift-docs/blame/main/installing/installing_aws/installing-aws-localzone.adoc#L80):

```
[role="_additional-resources"]
.Additional resources

* xref:../../networking/changing-cluster-network-mtu.adoc#mtu-value-selection_changing-cluster-network-mtu[MTU value selection]
```

Author: 👍🏽

<1> For `<installation_directory>`, specify the installation directory that
contains the `install-config.yaml` file you created.

> . Set the default Maximum Transmission Unit (MTU) according to the network plugin:

Reviewer: ```suggestion
. Set the default Maximum Transmission Unit (MTU) according to the network plugin:
+
```

[IMPORTANT]
====

> Generally, the Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300, see [AWS Documentation](https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html). 

Reviewer: Asciidoc links are formatted like this:
```suggestion
Generally, the Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300. See link:https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html[How Local Zones work] in the AWS documentation. 
```

The Cluster network MTU must be always less than the EC2 MTU to account for the cluster network overlay overhead. The specific overhead is determined by your network plugin:
- OVN-Kubernetes: 100 bytes
- OpenShift SDN: 50 bytes

Check the ['MTU value selection' for more details](https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu)
====

> > TODO: Steps described on the official doc https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu

Reviewer 2: :+1:

> > TODO: Do we need to add a pop-up explaining that it's a requirement of AWS to communicate withing the nodes in Local Zones and regular zones?

Reviewer 2: Agree. Because this is _nature_ limitation in different AWS zones

Author: Done!

<1> For `<installation_directory>`, specify the installation directory that
contains the `install-config.yaml` file you created.

. Set the default Maximum Transmission Unit (MTU) according to the network plugin:
+
[IMPORTANT]
====
Generally, the Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300, see [AWS Documentation](https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html). 
The Cluster network MTU must be always less than the EC2 MTU to account for the cluster network overlay overhead. The specific overhead is determined by your network plugin:
- OVN-Kubernetes: 100 bytes
- OpenShift SDN: 50 bytes

Check the ['MTU value selection' for more details](https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu)
====

> > TODO: Steps described on the official doc https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu
> 
> > TODO: Do we need to add a pop-up explaining that it's a requirement of AWS to communicate withing the nodes in Local Zones and regular zones?
> 
> > TODO: check the impact on cluster. Make sure we are validating this on the tests.

Author: ```suggestion
// Note: Steps described on the official doc https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu

// TODO: check the impact on cluster. Make sure we are validating this changes on e2e and QE tests.
```

<1> For `<installation_directory>`, specify the installation directory that
contains the `install-config.yaml` file you created.

. Set the default Maximum Transmission Unit (MTU) according to the network plugin:
+
[IMPORTANT]
====
Generally, the Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300. See link:https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html[How Local Zones work] in the AWS documentation. 

> The cluster network MTU must be always less than the EC2 MTU to account the overhead. The specific overhead is determined by your network plugin, for example:

Reviewer 3: Not sure if this should be "to account _for_ the overhead".

Author: Done, thanks!

- OVN-Kubernetes: `100 bytes`
- OpenShift SDN: `50 bytes`

The network plugin could provide additional features, like IPsec, that also must be decreased the MTU. Check the documentation for additional information.

====

.. If you are using the `OVN-Kubernetes` network plugin, enter the following command:
+
[source,terminal]
----

> $ cat <<EOF > <installation_directory>/manifests/cluster-network-03-config.yml

Author: @patrickdillon @yunjiang29 I just found that the installer is creating the file [`cluster-network-03-config.yml`](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/network.go#L21), the good news is it is only for P/Z platform. Do you think that we need to keep adding this `03` file or we must to change to `99`?

```suggestion
$ cat <<EOF > <installation_directory>/manifests/cluster-network-99-config.yml
```

Author: Closing this thread as it does not affect AWS, and phase1 we must have this fixed.

<1> For `<installation_directory>`, specify the installation directory that
contains the `install-config.yaml` file you created.

. Set the default Maximum Transmission Unit (MTU) according to the network plugin:
+
[IMPORTANT]
====
Generally, the Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300. See link:https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html[How Local Zones work] in the AWS documentation. 
The Cluster network MTU must be always less than the EC2 MTU to account for the cluster network overlay overhead. The specific overhead is determined by your network plugin:

> - OVN-Kubernetes: 100 bytes

Reviewer 4: OVN-IPsec overhead is an extra 46 bytes, this is be documented in the IPsec page `modules/nw-ovn-ipsec-enable.adoc`

I assume nobody is using IPsec with Local Zones.

Author: I avoid covering all variants here. I am assuming the user is being notified and should check the additional resources for more information. If you think that it is required to add here, please let me know.

- OpenShift SDN: 50 bytes

> [role="_additional-resources"]

Reviewer: This section needs to go in the assembly not the module.

Author: done! Thanks!

.Additional resources

> * xref:../../networking/changing-cluster-network-mtu.adoc#mtu-value-selection_changing-cluster-network-mtu[MTU value selection]

Reviewer 4: It looks like we don't have any documentation anywhere for minimum MTUs.   68 (or 576) for IPv4 and 1280 for IPv6.  I'll file a separate issue.

====

> // Note: Steps described on the official doc https://docs.openshift.com/container-platform/4.12/networking/changing-cluster-network-mtu.html#mtu-value-selection_changing-cluster-network-mtu

Author: This line should be removed after we decided if we will make any reference to this page or not
```suggestion
```

> // TODO: check the impact on cluster. Make sure we are validating this changes on e2e and QE tests.

Author: This line should be removed after we have the issue tracking the history or plans to run that performance change.
We also need to make sure we will not have issues in other components on the cluster, so we must check this before merging this PR
```suggestion
```

Author: QE team have been added more profiles to cover more scenarios in LZ on CI: https://github.com/openshift/release/pull/35952
Additional work has been tracked with this card: https://issues.redhat.com/browse/SPLAT-938
